projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/pojo/Environment.java

```diff
@@ -45,7 +46,8 @@ public Environment(
       @JsonProperty(PROP_NODE_BLACKLIST) @Nullable Set<String> nodeBlacklist,
       @JsonProperty(PROP_ENVIRONMENT_BGP_TABLES) @Nullable Map<String, String> bgpTables,
       @JsonProperty(PROP_ENVIRONMENT_ROUTING_TABLES) @Nullable Map<String, String> routingTables,
-      @JsonProperty(PROP_EXTERNAL_BGP_ANNOUNCEMENTS) @Nullable String externalBgpAnnouncements) {
+      @JsonProperty(PROP_EXTERNAL_BGP_ANNOUNCEMENTS) @Nullable
+          Set<BgpAdvertisement> externalBgpAnnouncements) {
```

Comment:
Should be SortedSet for consistent refs and to ease debugging. Generally, any Jackson collection should be SortedSet, SortedMap, or List for consistent ordering.

Reply:
Actually all the `Sets` and `Maps` (not just lines you modified) in this file should be switched to sorted versions if you don't mind.